internal/service/advancedclustertpf/resource_compatiblity.go

```diff
@@ -53,12 +63,23 @@ func resolveAPIInfo(ctx context.Context, plan *TFModel, diags *diag.Diagnostics,
 		diags.AddError("resolveContainerIDs failed", err.Error())
 		return nil
 	}
+	var (
+		legacySchema      bool
+		zoneNameNumShards map[string]int64
+	)
+	if overrideUsingLegacySchema {
+		legacySchema = true
+		zoneNameNumShards = numShardsMapFromOldAPI(clusterRespOld)
```

Comment:
plan is empty in Read when overriding legacy, so we get it from the old API

Reply:
Add this comment to the code?

Reply:
added in: https://github.com/mongodb/terraform-provider-mongodbatlas/pull/2896/commits/d39cf2f073eec031e73dc8f9b124673b5aca6855